src/textual/screen.py

```diff
@@ -307,6 +307,29 @@ def get_widgets_at(self, x: int, y: int) -> Iterable[tuple[Widget, Region]]:
         """
         return self._compositor.get_widgets_at(x, y)
 
+    def get_focusable_widget_at(self, x: int, y: int) -> Widget | None:
+        """Get the focusable widget under a given coordinate.
+
+        If the widget directly under the given coordinate is not focusable, then this method will check
+        if any of the ancestors are focusable. If no ancestors are focusable, then `None` will be returned.
+
+        Args:
+            x: X coordinate.
+            y: Y coordinate.
+
+        Returns:
+            A `Widget`, or `None` if there is no focusable widget underneath the coordinate.
+        """
+        try:
+            widget, _region = self.get_widget_at(x, y)
+        except NoWidget:
+            return None
+
+        for node in widget.ancestors_with_self:
+            if isinstance(node, Widget) and node.focusable:
+                return node
```

Comment:
That rarely occurs, since widgets are clipped within the boundary of their parents. The one exception is the overlay rule, and we decided that if you click on the overlay it should focus the DOM parent, even if it isn't directly under the mouse.